src/corva/handlers.py

```diff
@@ -544,3 +562,55 @@ def _get_custom_event_type_by_raw_aws_event(
         if events:
             return event_type, handler
     return None, None
+
+
+def _merge_events(aws_event: Any, data_transformation_type: Type[RawBaseEvent]) -> Any:
+    """
+    Merges incoming aws_events into one.
+    Merge happens differently, depending on app type.
+    Only "scheduled" and "stream" type of apps can be processed here.
+    If somehow any other type is passed - raise an exception
+    """
+    if data_transformation_type == RawScheduledEvent:
+        # scheduled event
+        if not isinstance(aws_event[0], dict):
+            aws_event = list(itertools.chain(*aws_event))
+        is_depth = aws_event[0]["scheduler_type"] == SchedulerType.data_depth_milestone
+        event_start, event_end = (
+            ("top_depth", "bottom_depth")
+            if is_depth
+            else ("schedule_start", "schedule_end")
+        )
+        min_event_start, max_event_end = (
+            aws_event[0][event_start],
+            aws_event[0].get(event_end),
+        )
+        for event in aws_event[1:]:
+            if event[event_start] < min_event_start:
+                min_event_start = event[event_start]
+            if (
+                max_event_end
+                and event.get(event_end)
+                and event[event_end] > max_event_end
+            ):
+                max_event_end = event[event_end]
+
+        aws_event[0][event_start] = min_event_start
+        if max_event_end:
+            aws_event[0][event_end] = max_event_end
+        aws_event = aws_event[0]
+        return aws_event
+
+    elif data_transformation_type == RawStreamEvent:
+        # stream event
+        for event in aws_event[1:]:
+            aws_event[0]["records"].extend(event["records"])
+        aws_event = [aws_event[0]]
+        return aws_event
+
+    # unexpected event type, raise an exception
+    raise RuntimeError(
+        "merge_events parameter was passed to app type other than scheduled "
+        "or stream. Merge strategy is not implemented for "
+        f"{data_transformation_type} event types."
+    )
```

Comment:
Wanted to share an interesting method on how to disallow passing other app types on Mypy static type check level. This will also probably make `test_events_not_merged_on_unexpected_event_type` obsolete
```python
if data_transformation_type is RawScheduledEvent:
    ...
elif data_transformation_type is RawStreamEvent:
    ...
else:
    typing.assert_never(data_transformation_type)
```

Reply:
Easier to just specify correct types in func definition `data_transformation_type: Union[Type[RawBaseEvent], ...]`